### PR TITLE
Add option for lights always on

### PR DESCRIPTION
**Pull request by dav0000000**

### CODE REVIEW

**Comment by aalm** [comment]
Seems reasonable to have a byte in config for the light setup(easy to extended use for other lighting schemes too).

**Comment by dav0000000** [comment]
Get your own byte. No, just joking. It was disappointing for me when the code began supporting the BBS02 and now the TSDZ2. It is kind of limited by the lowest common denominator now without the code getting too complicated. I can't think about the implications of how my changes affect the other motors and I'm certainly not worried about sharing a byte or other memory optimisation but I've still had to be careful. I use 5 sectors and the code compiles to 2712 bytes and still runs flawlessly on the BBSHD. The things I've added make my bike such a pleasure to ride. It's a nightmare for me to try to include any of my stuff back into this master as I'm just not a good enough programmer. I split off when the BBS02 came on board. It's complicated enough with just one display, battery voltage, motor and laws. It's coming up on two years that I've exclusively used this firmware. Every day I give respect to Daniel and you and many others have great talent too. This little community is awesome.

**Comment by danielnilsson9** [comment]
Thank you for this, I will review. I understand it can be a bit problematic for you since you had some own modifications since before the code was restructured and those might have to be manually worked into the new structure with bbsx/tsdz2 source directories to be integrated in master.

As long as the changes are in files outside the bbsx/tsdz2 directories you should not really have to think about which target they are for, should work for everybody.

BBSHD/BBS02/TSDZ2 all have more than 1kb of free RAM now so that is not an issue anymore. There is also a lot of the ROM left on all targets.

**Comment by dav0000000** [comment]
That's good to know about the memory. I thought it was still close to limits.

**Comment by danielnilsson9** [comment]
Do you want to make the necessary changes to this PR to have this merged or should I take over and integrate it to master?

**Comment by danielnilsson9** [comment]
Ah sorry @dav0000000, I did a review of this PR last week but looks like a never pressed publish...

**Review by danielnilsson9 — CHANGES_REQUESTED** [review]

**Comment by dav0000000** [comment]
Ok, I'll try to address the issues. Thanks.

File: src/tool/Model/Configuration.cs [hunk 1]
```diff
@@ -458,6 +466,76 @@ public bool ParseFromBufferV3(byte[] buffer)
...
+				MaxBatteryVolts = br.ReadUInt16() / 100f;
+				LowCutoffVolts = br.ReadByte();
+				MaxSpeedKph = br.ReadByte();
+
+				UseSpeedSensor = br.ReadBoolean();
+				UseShiftSensor = br.ReadBoolean();
+				UsePushWalk = br.ReadBoolean();
+				UseTemperatureSensor = (TemperatureSensor)br.ReadByte();
+
+				LightsAlwaysOn = br.ReadBoolean();
```
[the remark this answers is not in the document]
Comment (dav0000000) [reply src/tool/Model/Configuration.cs:498]: I've added LightsAlwaysOn = false to v1, v2 and v3.

File: tool/BBSFWTool.exe
[the remark this answers is not in the document]
Comment (dav0000000) [reply tool/BBSFWTool.exe:1]: Binary files have been removed.

File: src/firmware/bbsx/lights.c [hunk 2]
```diff
@@ -17,7 +18,7 @@ void lights_init()
 	SET_PIN_OUTPUT(PIN_LIGHTS);
 
 	lights_disable();
-	lights_set(false);
+	lights_set(g_config.lights_always_on);
```
[the remark this answers is not in the document]
Comment (dav0000000) [reply src/firmware/bbsx/lights.c]: Moved to app.init as requested.

File: src/firmware/bbsx/lights.c [hunk 3]
```diff
@@ -7,6 +7,7 @@
  */
 
 #include "lights.h"
+#include "cfgstore.h"
```
Comment (danielnilsson9) [inline src/firmware/bbsx/lights.c:10]: remove, not needded

File: src/firmware/tsdz2/lights.c [hunk 4]
```diff
@@ -18,7 +18,7 @@ void lights_init()
 	SET_PIN_OUTPUT(PIN_LIGHTS);
 
 	lights_enabled = false;
-	lights_set(false);
+	lights_set(g_config.lights_always_on);
```
Comment (danielnilsson9) [inline src/firmware/tsdz2/lights.c:21]: not sure what happened here, should be false, same as bbsx

**Review by danielnilsson9 — CHANGES_REQUESTED** [review]

File: src/firmware/bbsx/lights.c (hunk 3 above)
Comment (dav0000000) [reply src/firmware/bbsx/lights.c]: Done.

File: src/firmware/tsdz2/lights.c (hunk 4 above)
Comment (dav0000000) [reply src/firmware/tsdz2/lights.c]: Done

**Pull request merged**